### ISSUE TITLE
Training process

### ISSUE THREAD

**Issue by wangdaha1** [dongzelian/SSF#6]
Hi dongze,

When I try to run the script for stanfordcars dataset, the acc@1 for both Test and Test (EMA) are both 100% since epoch 2. 
( Test: [ 505/505]  Time: 0.049 (0.077)  Loss:  0.1048 (0.1048)  Acc@1: 100.0000 (100.0000)  Acc@5: 100.0000 (100.0000)
 Test (EMA): [ 505/505]  Time: 0.049 (0.077)  Loss:  0.1050 (0.1050)  Acc@1: 100.0000 (100.0000)  Acc@5: 100.0000 (100.0000)
*** Best metric: 100.0 (epoch 2))

The only differerce is that I run it on single GPU, and I delete `--pretrained` and add `--initial-checkpoint ./pretrained_models/imagenet21k_ViT-B_16.npz` (where I downloaded and uploaded the pretrained ViT model) in `train_ssf.sh`. Other parameters are kept the same. As for the stanfordcars dataset, I have use it to run VPT where everything is reasonable.

Did I make any mistake for running the script? Thanks for your help!

Best,

**Comment by dongzelian** [comment]
@wangdaha1 Strange accuracy. Is it normal for other datasets? Please check the dataset split.

**Comment by wangdaha1** [comment]
I have checked other datasets (stanforddogs, cub, nabirds) and they are reasonable while oxfordflowers and stanfordcars are not. I also found that in your `data/dataset_factory.py`, lines 152-170, there are dogs(), NABIRDS() and Cub2011() functions for processing the corresponding datasets while none for oxfordflowers and stanfordcars. I think this may be the reason.

Hence, I assume there may be some files missing in my datasets. However, I downloaded them from the official website. Could you please tell me the filenames that included in your oxfordflowers and stanfordcars datasets?

Thanks in advance!

**Comment by MouxiaoHuang** [comment]
@wangdaha1 Hi, have you solved the problem? I encountered the similar problem, too. I checked Oxford_Flowers and Stanford_Cars, the Acc@1 for Test and Test (EMA) of Stanford_Cars are also 100%.

I use `--pretrained` and the pre-trained model is downloaded from default url link. I run it on 2 GPUs and other parameters are kept the same.

My training script is as below,
`CUDA_VISIBLE_DEVICES=2,3,  python3  -m torch.distributed.launch --nproc_per_node=2  --master_port=12329  \
	train.py ./database/FGVC/Stanford_Cars --dataset stanford_cars --num_classes 196 --val_split val  --simple_aug --model vit_base_patch16_224_in21k  \
    --batch_size 32 --epochs 100 \
	--opt adamw  --weight_decay 0.05 \
    --warmup_lr 1e-7 --warmup_epochs 10  \
    --lr 2e-2 --min_lr 1e-8 \
    --drop_path 0 --img_size 224 \
	--model_ema --model_ema_decay 0.9998  \
	--output  output/vit_base_patch16_224_in21k/fgvc/stanford_cars/ssf \
	--amp --tuning_mode ssf --pretrained  
`

Here is my summary file of Stanford_Cars: [link](https://drive.google.com/file/d/1UaKx6uqJ2vj1vVO2a2lc2ju4LhwYgYds/view?usp=sharing)

Thanks and looking for your reply!

**Comment by MouxiaoHuang** [comment]
@dongzelian Thanks for your excellent work and contributions to this community first!

And I have some questions for the code. `dataset_factory.py` doesn't define Oxford_Flowers and Stanford_Cars datasets, and there're no `oxford_flowers.py` and `stanford_cars.py` in `data/` directory just like other three datasets in FGVC. Did I make mistakes or some files are missed in this repo?

**Comment by wangdaha1** [comment]
> @wangdaha1 Hi, have you solved the problem? I encountered the similar problem, too. I checked Oxford_Flowers and Stanford_Cars, the Acc@1 for Test and Test (EMA) of Stanford_Cars are also 100%.
> 
> I use `--pretrained` and the pre-trained model is downloaded from default url link. I run it on 2 GPUs and other parameters are kept the same.
> 
> My training script is as below, `CUDA_VISIBLE_DEVICES=2,3, python3 -m torch.distributed.launch --nproc_per_node=2 --master_port=12329 \ train.py ./database/FGVC/Stanford_Cars --dataset stanford_cars --num_classes 196 --val_split val --simple_aug --model vit_base_patch16_224_in21k \ --batch_size 32 --epochs 100 \ --opt adamw --weight_decay 0.05 \ --warmup_lr 1e-7 --warmup_epochs 10 \ --lr 2e-2 --min_lr 1e-8 \ --drop_path 0 --img_size 224 \ --model_ema --model_ema_decay 0.9998 \ --output output/vit_base_patch16_224_in21k/fgvc/stanford_cars/ssf \ --amp --tuning_mode ssf --pretrained `
> 
> Here is my summary file of Stanford_Cars: [link](https://drive.google.com/file/d/1UaKx6uqJ2vj1vVO2a2lc2ju4LhwYgYds/view?usp=sharing)
> 
> Thanks and looking for your reply!

Same problem and unsolved yet.

**Comment by dongzelian** [comment]
@wangdaha1 @MouxiaoHuang I have checked Oxford_Flowers and Stanford_Cars datasets, for which I use 'train/test' format instead of the original '.mat' format. You have two methods to solve this problem: i) reorganize the data file format through the code https://github.com/facebookresearch/vissl/blob/main/extra_scripts/datasets/create_oxford_flowers_data_files.py and https://github.com/facebookresearch/vissl/blob/main/extra_scripts/datasets/create_stanford_cars_data_files.py. I forget the specific github code I use to process both datasets but the above codes look work well. ii) use the dataset format following VPT https://github.com/KMnP/vpt/blob/main/src/data/datasets/json_dataset.py. You need to rewrite the oxfordflowers and stanfordcars classes. 

Thanks.
